packages/hint/src/lib/cli/analyze.ts

```diff
@@ -188,8 +188,14 @@ const loadUserConfig = async (actions?: CLIOptions): Promise<UserConfig> => {
         userConfig = getDefaultConfiguration();
     }
 
-    // If language is not in the config file, get the language configure in the OS.
-    userConfig.language = userConfig.language || await osLocale();
+    // If language is not in the config file or CLI options, get the language configure in the OS.
+    if (actions && actions.language) {
```

Comment:
Added this `if-else` to keep debug log entry, we were already having at other places from where `language` overriding code is removed.

Reply:
I like this, maybe we should have the same for the other two options. Also, maybe you can move the logic to their own `getLanguage` method.

Reply:
By other two options you `formatters` and `hints` right?

Reply:
I think if we move `updateConfigWithOptionsValues` to this place. It should work. All `debug` and all CLI options overriding logic will remain in one function.

Reply:
Sorry, I didn't explain myself very well, for other two options I meant add a debug message when we are using the language from the userconfig or the language using `osLocale`.

Reply:
Oh ok. Got it. Thanks

Reply:
@sarvaje made the change :)